wasm/src/error.rs

```diff
@@ -0,0 +1,52 @@
+use leb128::read;
+
+pub type Result<T> = std::result::Result<T, crate::error::Error>;
+
+#[repr(u8)]
+#[derive(Debug)]
+pub enum Error {
+    IoError(std::io::Error),
+    ReadZeroBytes,
+    TooManyBytes(u8),
```

Comment:
On its own, I don't think it's obvious as to what the u8 represents here?
As in, it's not clear as to whether it was the expected amount, or how many were read, even though I'd expect the latter to be usize.

Reply:
Should I add a comment? something like `// TooManyBytes(expected)`

Reply:
Could be `TooManyBytes { expected_bytes: u8 }`

Reply:
Sgtm